packages/apps/spaces/components/contact/contact-history/ContactHistory.tsx

```diff
@@ -15,37 +17,78 @@ export const ContactHistory = ({ id }: { id: string }) => {
     loading: ticketsLoading,
     error: ticketsError,
   } = useContactTickets({ id });
-  // const {data: actions, loading: actionsLoading, error: actionsError} = useContactActions({id});
   // TODO add pagination support
   const {
     data: conversations,
     loading: conversationsLoading,
     error: conversationsError,
   } = useContactConversations({ id });
 
+  const query = gql`query GetActionsForContact($id: ID!, $from: Time!, $to: Time! ) {
```

Comment:
Why do you want to do that this way? Please use codegen to generate types and query hooks. If you need to do request 'on demand' eg. on click or inside useEffect use 'useLazyQuery' (it is generated alongside the default one)

Reply:
the code generated by the codegen doesn't work. it produces an infinite loop in loading.